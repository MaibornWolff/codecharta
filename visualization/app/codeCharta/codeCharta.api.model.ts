import { AttributeTypes, AttributeTypeValue, CodeMapNode, Edge, MarkedPackage, RecursivePartial, Settings } from "./codeCharta.model"

export interface ExportCCFile {
	projectName: string
	apiVersion: string
	nodes: CodeMapNode[]
	attributeTypes?: AttributeTypes | OldAttributeTypes
	edges?: Edge[]
	markedPackages?: MarkedPackage[]
	blacklist?: ExportBlacklistItem[]
}

export interface ExportBlacklistItem {
	path: string
	type: ExportBlacklistType
}

export enum ExportBlacklistType {
	hide = "hide",
	exclude = "exclude"
}

export enum APIVersions {
	ZERO_POINT_ONE = "0.1",
	ONE_POINT_ZERO = "1.0",
	ONE_POINT_ONE = "1.1"
}

export interface ExportScenario {
	name: string
	settings: RecursivePartial<Settings>
}

export interface OldAttributeTypes {
	nodes?: [{ [key: string]: AttributeTypeValue }]
	edges?: [{ [key: string]: AttributeTypeValue }]
}
