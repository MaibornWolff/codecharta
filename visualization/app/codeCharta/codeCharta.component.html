<tool-bar-component></tool-bar-component>

<file-extension-bar-component></file-extension-bar-component>
<div class="ribbon-bar-wrapper">
	<ribbon-bar-component class="ribbon-bar-component"></ribbon-bar-component>
</div>

<code-map-component></code-map-component>

<legend-panel-component></legend-panel-component>

<loading-gif-file-component></loading-gif-file-component>

<div id="mw-logo">
	<div class="logo">
		<a href="https://www.maibornwolff.de" target="_blank"><img src="assets/logo.png" alt="MaibornWolff"/></a>
		<h2>
			<span>CodeCharta {{ $ctrl._viewModel.version }}</span>
		</h2>
	</div>
</div>

<node-context-menu-component></node-context-menu-component>
