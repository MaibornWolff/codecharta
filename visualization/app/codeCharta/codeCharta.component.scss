@import "../mixins.scss";

code-charta-component {
	#mw-logo {
		.logo {
			width: 160px;
			height: 100px;
			position: absolute;
			left: 0;
			bottom: 30px;
			z-index: 29;
		}

		a,
		img {
			width: 5em;
			display: block;
			margin: auto !important;
		}

		h2 {
			text-transform: uppercase;
			color: #5a585a;
			font-weight: 500;
			font-size: 9pt;
			text-align: center;
			margin: 0;
			z-index: -1;
		}
	}

	#codeMap {
		position: fixed;
		z-index: 0;
		top: 98px;
		left: 0;
		width: 100%;
		height: 100%;
	}

	.cc-shadow {
		box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2) !important;
	}

	.ribbon-bar-wrapper {
		height: 500px;
		position: fixed;
		z-index: 10;
		white-space: nowrap;
		pointer-events: none;
	}

	attribute-side-bar-component .side-bar-container,
	legend-panel-component .block-wrapper,
	legend-panel-component .panel-button,
	view-cube-component {
		transition: right 0.25s ease;
	}

	.angular-green {
		color: #00bfa5;
	}

	.small-action-button {
		display: inline-block;
		height: 24px;
		width: 24px;
		vertical-align: top;

		.md-fab {
			min-height: 0;
			width: 100%;
			height: 100%;
			margin: 0;

			i {
				position: absolute;
				top: 5px;
				left: 6px;
			}
		}

		&.transparent .md-fab {
			background-color: transparent !important;
			box-shadow: none;

			&:hover {
				background-color: #ddd !important;
			}
		}
	}

	.md-button.round-button {
		margin-right: 0 !important;
		float: right;
		min-width: 36px;
		width: 36px;
		border-radius: 50%;
		margin: 0;
	}

	.action-buttons {
		position: fixed;
		bottom: 2em;
		right: 2em;
		z-index: 60;

		.long-button {
			width: inherit;
			padding-left: 10px;
			padding-right: 10px;
			border-radius: 100px;
		}

		> * {
			margin-left: 4px;
			margin-right: 4px;
			z-index: 60;
		}
	}

	.state-selector-button {
		height: 22px;
		vertical-align: bottom;
		width: 30px;
		min-height: 0;
		min-width: 0;
		margin: 0;
		padding: 0;
		font-size: 9pt;
		text-transform: none;
		background-color: rgba(230, 230, 230, 0.96);

		&:hover {
			background-color: rgb(200, 200, 200) !important;
		}

		i {
			vertical-align: middle;
			height: 26px;
		}

		&.left {
			border-radius: 5px 0 0 5px;
		}

		&.middle {
			border-radius: 0;
			border-left: 1px solid rgba(255, 255, 255, 1);
			border-right: 1px solid rgba(255, 255, 255, 1);
		}

		&.right {
			border-radius: 0 5px 5px 0;
		}

		&.current {
			background-color: rgb(0, 150, 136);
			color: #ffffff;
		}

		&.current:hover {
			background-color: rgb(0, 124, 112) !important;
		}
	}
}
