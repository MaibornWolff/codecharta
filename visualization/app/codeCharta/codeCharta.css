*, *:before, *:after {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
}

html, body {
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    width: 100%;
    height: 100%;
    margin: 0;
    padding: 0;
    overflow: hidden;
    font-size: 1em;
    cursor: default;
    color: #5A585A;
    font-family: Roboto, sans-serif;
    font-weight: 300;
}

#codeMap {
    position: absolute;
    z-index: 0;
    top: 50px;
    left: 0;
    width: 100%;
    height: 100%;
}

#hud {
    position: absolute;
    z-index: 10;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    max-width: none !important;
    pointer-events: none;
    padding-left:10px;
    padding-right:10px;
}

revision-chooser-component, legend-panel-component, file-chooser-panel-directive, settings-panel-component, .mwlink {
    pointer-events: auto;
}

#title {
    text-transform: uppercase;
    font-size: 0.9em;
    line-height: 1.4em;
    float: right;
    margin: 0.5em 1.6em 0 0;
    color: #5A585A;
    font-family: Roboto, sans-serif;
    font-weight: 300;
}

#title:first-child {
    font-size: 1.82em;
}

#title:last-child {
    font-size: 1.5em;
}

.collapsible > li {
    background-color: #ffffff !important;
}

.col-backdrop {
    background-color: rgba(0,0,0,0.1);
}

.bottom-row {
    position: fixed;
    bottom: 0;
    width:100%;
}

.mwlink > img{
    width: 100%;
    max-width: 4em;
    float:right;
}

.chip {
    background-color: #efefef;
}

.row, .col {
    min-height: 0px !important;
}


/*

#actionButtons {
    position: fixed;
    bottom: 2em;
    right: 2em;
    z-index: 1000;
}

#settingsPanel {
    position: fixed;
    right: 0;
    width: 0px;
    height: 100%;
    background: #fff;
    z-index: 1000;
    box-shadow: 0 0 10px #000;
}

#detailWindow {
    position: fixed;
    z-index: 999;
    bottom: 0.5em;
    left: 0.5em;
    color:#5A585A;
    font-size: 0.8em;
}

#legendButton {
    position: absolute;
    top: 11.5em;
    left: -2.9em;
    -webkit-transform: rotate(90deg);
    -moz-transform: rotate(90deg);
    -o-transform: rotate(90deg);
    -ms-transform: rotate(90deg);
    transform: rotate(90deg);
}

#legendPanel {
    position: fixed;
    top: 9em;
    left: -500px;
}

#revisionButton {
    position: absolute;
    top: 21em;
    left: -3.5em;
    -webkit-transform: rotate(90deg);
    -moz-transform: rotate(90deg);
    -o-transform: rotate(90deg);
    -ms-transform: rotate(90deg);
    transform: rotate(90deg);
}

#mapButton {
    position: absolute;
    top: 21em;
    left: -2.3em;
    -webkit-transform: rotate(90deg);
    -moz-transform: rotate(90deg);
    -o-transform: rotate(90deg);
    -ms-transform: rotate(90deg);
    transform: rotate(90deg);
}

#revisionChooser {
    position: fixed;
    top: 18.5em;
    left: -500px;
}

.rzslider .rz-pointer, .rzslider .rz-selection {
    background: #26a69a !important;
}

.rzslider .rz-bar {
    background: rgb(196,196,196);
}*/