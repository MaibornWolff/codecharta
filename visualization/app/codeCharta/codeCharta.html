<div id="hud" class="container">

    <div class="row">

        <div class="col l3 m6 s12 col-backdrop">
            <legend-panel-component></legend-panel-component>
        </div>

        <div class="col l3 m6 s12 col-backdrop">
            <revision-chooser-component></revision-chooser-component>
        </div>

        <div class="col l3 m6 s12 col-backdrop">
            <file-chooser-panel-directive></file-chooser-panel-directive>
        </div>

        <div class="col l3 m6 s12 col-backdrop">
            <settings-panel-component></settings-panel-component>
        </div>

    </div>

    <div class="row bottom-row">

        <div class="col l3 m6 s12 col-backdrop">
            <detail-panel-component></detail-panel-component>
        </div>

    </div>

    <div id="titlebox">
        <a class="mwlink" href="https://www.maibornwolff.de" target="_blank"><img src="./logo.png" alt="MaibornWolff"></a>
        <h1 id="title"><span>CodeCharta {{$ctrl.pkg.version}}</span></h1>
    </div>

</div>

<code-map-component></code-map-component>