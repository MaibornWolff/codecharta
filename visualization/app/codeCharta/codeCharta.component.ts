import { UrlExtractor } from "./util/urlExtractor"
import { IHttpService, ILocationService } from "angular"
import "./codeCharta.component.scss"
import { CodeChartaService } from "./codeCharta.service"
import { ScenarioHelper } from "./util/scenarioHelper"
import { DialogService } from "./ui/dialog/dialog.service"
import { NameDataPair } from "./codeCharta.model"
import { InjectorService } from "./state/injector.service"
import { StoreService } from "./state/store.service"
import { setState } from "./state/store/state.actions"
import { setAppSettings } from "./state/store/appSettings/appSettings.actions"
import { setIsLoadingFile } from "./state/store/appSettings/isLoadingFile/isLoadingFile.actions"
import * as codeCharta from "../../package.json"
import { setDelta, setMultiple, setSingle } from "./state/store/files/files.actions"

export class CodeChartaController {
	private _viewModel: {
		version: string
	} = {
		version: "version unavailable"
	}

	private urlUtils: UrlExtractor

	/* @ngInject */
	constructor(
		private $location: ILocationService,
		private $http: IHttpService,
		private storeService: StoreService,
		private dialogService: DialogService,
		private codeChartaService: CodeChartaService,
		// @ts-ignore
		private injectorService: InjectorService // We have to inject it somewhere
	) {
		this._viewModel.version = codeCharta.version
		this.urlUtils = new UrlExtractor(this.$location, this.$http)
		this.storeService.dispatch(setIsLoadingFile(true))
		this.loadFileOrSample()
	}

	public loadFileOrSample() {
		return this.urlUtils
			.getFileDataFromQueryParam()
			.then((data: NameDataPair[]) => {
				if (data.length > 0) {
					this.tryLoadingFiles(data)
					this.setRenderStateFromUrl()
				} else {
					this.tryLoadingSampleFiles()
				}
			})
			.catch(() => {
				this.tryLoadingSampleFiles()
			})
	}

	public tryLoadingSampleFiles() {
		if (this.urlUtils.getParameterByName("file")) {
			this.dialogService.showErrorDialog(
				"One or more files from the given file URL parameter could not be loaded. Loading sample files instead."
			)
		}
		this.tryLoadingFiles([
			{ fileName: "sample1.cc.json", content: require("./assets/sample1.cc.json") },
			{ fileName: "sample2.cc.json", content: require("./assets/sample2.cc.json") }
		])
	}

	private tryLoadingFiles(values: NameDataPair[]) {
		this.storeService.dispatch(setAppSettings())

		this.codeChartaService
			.loadFiles(values)
			.then(() => {
				this.storeService.dispatch(setState(ScenarioHelper.getDefaultScenario().settings))
			})
			.catch(e => {
				this.storeService.dispatch(setIsLoadingFile(false))
				console.error(e)
				this.printErrors(e)
			})
	}

	private setRenderStateFromUrl() {
		const renderState: string = this.urlUtils.getParameterByName("mode")
		const files = this.storeService.getState().files.getCCFiles()

		if (renderState === "Delta" && files.length >= 2) {
			this.storeService.dispatch(setDelta(files[0], files[1]))
		} else if (renderState === "Multiple") {
			this.storeService.dispatch(setMultiple(files))
		} else {
			this.storeService.dispatch(setSingle(files[0]))
		}
	}

	private printErrors(errors: Object) {
		this.dialogService.showErrorDialog(JSON.stringify(errors, null, "\t"))
	}
}

export const codeChartaComponent = {
	selector: "codeChartaComponent",
	template: require("./codeCharta.component.html"),
	controller: CodeChartaController
}
