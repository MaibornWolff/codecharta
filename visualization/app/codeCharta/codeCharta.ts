"use strict";

import "./core/core.module";
import "./ui/ui";

import {codeChartaComponent, CodeChartaController} from "./codeCharta.component";

import angular from "angular";

angular.module(
    "app.codeCharta",
    ["app.codeCharta.core", "app.codeCharta.ui"]
);

angular.module("app.codeCharta").component(
    codeChartaComponent.selector,
    codeChartaComponent
);


