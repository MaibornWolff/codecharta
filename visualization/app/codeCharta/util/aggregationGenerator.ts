import { CodeMapNode, CCFile, NodeType } from "../codeCharta.model"
import { CodeChartaService } from "../codeCharta.service"
import { FileNameHelper } from "./fileNameHelper"
import { getUpdatedPath } from "./nodePathHelper"
import packageJson from "../../../package.json"

export class AggregationGenerator {
	private static projectNameArray: string[] = []
	private static fileNameArray: string[] = []

	static getAggregationFile(inputFiles: CCFile[]) {
		if (inputFiles.length === 1) {
			return inputFiles[0]
		}

		this.resetVariables()

		for (const inputFile of inputFiles) {
			this.projectNameArray.push(inputFile.fileMeta.projectName.replace(" ", "_"))
			this.fileNameArray.push(FileNameHelper.withoutCCJsonExtension(inputFile.fileMeta.fileName).replace(" ", "_"))
		}
		return this.getNewAggregatedMap(inputFiles)
	}

	private static getNewAggregatedMap(inputFiles: CCFile[]) {
		const outputFile: CCFile = {
			fileMeta: {
				projectName: `project_aggregation_of_${this.projectNameArray.join("_and_")}`,
				fileName: `file_aggregation_of_${this.fileNameArray.join("_and_")}`,
				apiVersion: packageJson.codecharta.apiVersion
			},
			map: {
				name: CodeChartaService.ROOT_NAME,
				type: NodeType.FOLDER,
				children: [],
				attributes: {},
				path: CodeChartaService.ROOT_PATH
			},
			settings: {
				fileSettings: {
					edges: [],
					blacklist: [],
					attributeTypes: { nodes: {}, edges: {} },
					markedPackages: []
				}
			}
		}

		for (const inputMap of inputFiles) {
			outputFile.map.children.push(this.extractNodeFromMap(inputMap))
		}
		this.aggregateRootAttributes(outputFile)

		return outputFile
	}

	private static aggregateRootAttributes(outputFile: CCFile) {
		for (const child of outputFile.map.children) {
			const { attributes } = child
			// TODO: Remove `in` usage (in general throughout the code)
			for (const key in attributes) {
				if (!(key in outputFile.map.attributes)) {
					outputFile.map.attributes[key] = 0
				}
				outputFile.map.attributes[key] += attributes[key]
			}
		}
	}

	private static extractNodeFromMap(inputMap: CCFile) {
		const outputNode: CodeMapNode = {
			name: inputMap.fileMeta.fileName,
			children: inputMap.map.children,
			type: inputMap.map.type
		}

		if (inputMap.map.path) {
			outputNode.path = getUpdatedPath(inputMap.fileMeta.fileName, inputMap.map.path)
		}

		for (const key in inputMap.map) {
			if (!["name", "path", "children"].includes(key)) {
				outputNode[key] = inputMap.map[key]
			}
		}
		this.updatePathOfAllChildren(inputMap.fileMeta.fileName, outputNode.children)
		return outputNode
	}

	private static updatePathOfAllChildren(fileName: string, children: CodeMapNode[]) {
		for (const child of children) {
			if (child.path) {
				child.path = getUpdatedPath(fileName, child.path)
			}

			if (child.children) {
				this.updatePathOfAllChildren(fileName, child.children)
			}
		}
	}

	private static resetVariables() {
		this.projectNameArray = []
		this.fileNameArray = []
	}
}
