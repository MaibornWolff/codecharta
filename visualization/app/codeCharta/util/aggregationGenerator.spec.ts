import { CCFile, NodeType, Settings } from "../codeCharta.model"
import { AggregationGenerator } from "./aggregationGenerator"
import packageJson from "../../../package.json"

describe("AggregationGenerator", () => {
	const file1: CCFile = {
		fileMeta: {
			fileName: "file1",
			projectName: "Sample Project",
			apiVersion: packageJson.codecharta.apiVersion
		},
		map: {
			name: "root",
			type: NodeType.FOLDER,
			path: "/root",
			attributes: { rloc: 170, functions: 1010, mcc: 11 },
			children: [
				{
					name: "big leaf",
					type: NodeType.FILE,
					path: "/root/big leaf",
					attributes: { rloc: 100, functions: 10, mcc: 1 },
					link: "http://www.google.de"
				},
				{
					name: "Parent Leaf",
					type: NodeType.FOLDER,
					path: "/root/Parent Leaf",
					attributes: { rloc: 70, functions: 1000, mcc: 10 },
					children: [
						{
							name: "other small leaf",
							type: NodeType.FILE,
							path: "/root/Parent Leaf/other small leaf",
							attributes: { rloc: 70, functions: 1000, mcc: 10 }
						}
					]
				}
			]
		},
		settings: {} as Settings
	}

	const file2: CCFile = {
		fileMeta: {
			fileName: "file2",
			projectName: "Sample Project",
			apiVersion: packageJson.codecharta.apiVersion
		},
		map: {
			name: "root",
			type: NodeType.FOLDER,
			path: "/root",
			attributes: { rloc: 260, functions: 220, mcc: 202, customMetric: 7 },
			children: [
				{
					name: "big leaf",
					type: NodeType.FILE,
					path: "/root/big leaf",
					attributes: { rloc: 200, functions: 20, mcc: 2 },
					link: "http://www.google.de"
				},
				{
					name: "Parent Leaf",
					type: NodeType.FOLDER,
					path: "/root/Parent Leaf",
					attributes: { rloc: 60, functions: 200, mcc: 200 },
					children: [
						{
							name: "small leaf",
							type: NodeType.FILE,
							path: "/root/Parent Leaf/small leaf",
							attributes: { rloc: 60, functions: 200, mcc: 200 }
						}
					]
				}
			]
		},
		settings: {} as Settings
	}

	describe("multipleService", () => {
		it("aggregation of two maps", () => {
			const aggregationFile = AggregationGenerator.getAggregationFile([file1, file2])
			expect(aggregationFile).toMatchSnapshot()
		})

		it("aggregation of four maps", () => {
			const aggregationFile = AggregationGenerator.getAggregationFile([file1, file2, file1, file2])
			expect(aggregationFile).toMatchSnapshot()
		})

		it("aggregation one map", () => {
			const aggregationFile = AggregationGenerator.getAggregationFile([file1])
			expect(aggregationFile).toMatchSnapshot()
		})

		it("aggregate two aggregated maps should aggregate the attributes to root", () => {
			const aggregationFile = AggregationGenerator.getAggregationFile([file1, file2])
			expect(aggregationFile.map.attributes.rloc).toBe(430)
			expect(aggregationFile.map.attributes.functions).toBe(1230)
			expect(aggregationFile.map.attributes.mcc).toBe(213)
			expect(aggregationFile.map.attributes.customMetric).toBe(7)
		})
	})
})
