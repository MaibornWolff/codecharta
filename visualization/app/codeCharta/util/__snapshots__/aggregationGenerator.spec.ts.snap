// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregationGenerator multipleService aggregation of four maps 1`] = `
Object {
  "fileMeta": Object {
    "apiVersion": "1.2",
    "fileChecksum": "md5-file1;md5-file2;md5-file1;md5-file2;md5-file1;md5-file2",
    "fileName": "file_aggregation_of_file1_and_file2_and_file1_and_file2",
    "projectName": "project_aggregation_of_Sample_Project_and_Sample_Project_and_Sample_Project_and_Sample_Project",
  },
  "map": Object {
    "attributes": Object {
      "customMetric": 14,
      "functions": 2460,
      "mcc": 426,
      "rloc": 860,
    },
    "children": Array [
      Object {
        "attributes": Object {
          "functions": 1010,
          "mcc": 11,
          "rloc": 170,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 10,
              "mcc": 1,
              "rloc": 100,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file1/file1/file1/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 1000,
              "mcc": 10,
              "rloc": 70,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 1000,
                  "mcc": 10,
                  "rloc": 70,
                },
                "name": "other small leaf",
                "path": "/root/file1/file1/file1/Parent Leaf/other small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file1/file1/file1/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file1",
        "path": "/root/file1",
        "type": "Folder",
      },
      Object {
        "attributes": Object {
          "customMetric": 7,
          "functions": 220,
          "mcc": 202,
          "rloc": 260,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 20,
              "mcc": 2,
              "rloc": 200,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file2/file2/file2/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 200,
              "mcc": 200,
              "rloc": 60,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 200,
                  "mcc": 200,
                  "rloc": 60,
                },
                "name": "small leaf",
                "path": "/root/file2/file2/file2/Parent Leaf/small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file2/file2/file2/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file2",
        "path": "/root/file2",
        "type": "Folder",
      },
      Object {
        "attributes": Object {
          "functions": 1010,
          "mcc": 11,
          "rloc": 170,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 10,
              "mcc": 1,
              "rloc": 100,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file1/file1/file1/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 1000,
              "mcc": 10,
              "rloc": 70,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 1000,
                  "mcc": 10,
                  "rloc": 70,
                },
                "name": "other small leaf",
                "path": "/root/file1/file1/file1/Parent Leaf/other small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file1/file1/file1/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file1",
        "path": "/root/file1",
        "type": "Folder",
      },
      Object {
        "attributes": Object {
          "customMetric": 7,
          "functions": 220,
          "mcc": 202,
          "rloc": 260,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 20,
              "mcc": 2,
              "rloc": 200,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file2/file2/file2/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 200,
              "mcc": 200,
              "rloc": 60,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 200,
                  "mcc": 200,
                  "rloc": 60,
                },
                "name": "small leaf",
                "path": "/root/file2/file2/file2/Parent Leaf/small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file2/file2/file2/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file2",
        "path": "/root/file2",
        "type": "Folder",
      },
    ],
    "name": "root",
    "path": "/root",
    "type": "Folder",
  },
  "settings": Object {
    "fileSettings": Object {
      "attributeTypes": Object {
        "edges": Object {},
        "nodes": Object {},
      },
      "blacklist": Array [],
      "edges": Array [],
      "markedPackages": Array [],
    },
  },
}
`;

exports[`AggregationGenerator multipleService aggregation of two maps 1`] = `
Object {
  "fileMeta": Object {
    "apiVersion": "1.2",
    "fileChecksum": "md5-file1;md5-file2",
    "fileName": "file_aggregation_of_file1_and_file2",
    "projectName": "project_aggregation_of_Sample_Project_and_Sample_Project",
  },
  "map": Object {
    "attributes": Object {
      "customMetric": 7,
      "functions": 1230,
      "mcc": 213,
      "rloc": 430,
    },
    "children": Array [
      Object {
        "attributes": Object {
          "functions": 1010,
          "mcc": 11,
          "rloc": 170,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 10,
              "mcc": 1,
              "rloc": 100,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file1/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 1000,
              "mcc": 10,
              "rloc": 70,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 1000,
                  "mcc": 10,
                  "rloc": 70,
                },
                "name": "other small leaf",
                "path": "/root/file1/Parent Leaf/other small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file1/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file1",
        "path": "/root/file1",
        "type": "Folder",
      },
      Object {
        "attributes": Object {
          "customMetric": 7,
          "functions": 220,
          "mcc": 202,
          "rloc": 260,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 20,
              "mcc": 2,
              "rloc": 200,
            },
            "link": "http://www.google.de",
            "name": "big leaf",
            "path": "/root/file2/big leaf",
            "type": "File",
          },
          Object {
            "attributes": Object {
              "functions": 200,
              "mcc": 200,
              "rloc": 60,
            },
            "children": Array [
              Object {
                "attributes": Object {
                  "functions": 200,
                  "mcc": 200,
                  "rloc": 60,
                },
                "name": "small leaf",
                "path": "/root/file2/Parent Leaf/small leaf",
                "type": "File",
              },
            ],
            "name": "Parent Leaf",
            "path": "/root/file2/Parent Leaf",
            "type": "Folder",
          },
        ],
        "name": "file2",
        "path": "/root/file2",
        "type": "Folder",
      },
    ],
    "name": "root",
    "path": "/root",
    "type": "Folder",
  },
  "settings": Object {
    "fileSettings": Object {
      "attributeTypes": Object {
        "edges": Object {},
        "nodes": Object {},
      },
      "blacklist": Array [],
      "edges": Array [],
      "markedPackages": Array [],
    },
  },
}
`;

exports[`AggregationGenerator multipleService aggregation one map 1`] = `
Object {
  "fileMeta": Object {
    "apiVersion": "1.2",
    "fileChecksum": "md5-file1",
    "fileName": "file1",
    "projectName": "Sample Project",
  },
  "map": Object {
    "attributes": Object {
      "functions": 1010,
      "mcc": 11,
      "rloc": 170,
    },
    "children": Array [
      Object {
        "attributes": Object {
          "functions": 10,
          "mcc": 1,
          "rloc": 100,
        },
        "link": "http://www.google.de",
        "name": "big leaf",
        "path": "/root/file1/file1/file1/big leaf",
        "type": "File",
      },
      Object {
        "attributes": Object {
          "functions": 1000,
          "mcc": 10,
          "rloc": 70,
        },
        "children": Array [
          Object {
            "attributes": Object {
              "functions": 1000,
              "mcc": 10,
              "rloc": 70,
            },
            "name": "other small leaf",
            "path": "/root/file1/file1/file1/Parent Leaf/other small leaf",
            "type": "File",
          },
        ],
        "name": "Parent Leaf",
        "path": "/root/file1/file1/file1/Parent Leaf",
        "type": "Folder",
      },
    ],
    "name": "root",
    "path": "/root",
    "type": "Folder",
  },
  "settings": Object {},
}
`;
